src/askui/models/ui_tars_ep/ui_tars_api.py

```diff
@@ -201,24 +203,13 @@ def execute_act(self, message_history):
             return
 
         action = message.parsed_action
-        if action.action_type == "click":
-            self._agent_os.mouse(action.start_box.x, action.start_box.y)
-            self._agent_os.click("left")
-            time.sleep(1)
-        if action.action_type == "type":
-            self._agent_os.click("left")
-            self._agent_os.type(action.content)
-            time.sleep(0.5)
-        if action.action_type == "hotkey":
-            self._agent_os.keyboard_pressed(action.key)
-            self._agent_os.keyboard_release(action.key)
-            time.sleep(0.5)
-        if action.action_type == "call_user":
-            time.sleep(1)
-        if action.action_type == "wait":
-            time.sleep(2)
-        if action.action_type == "finished":
+        if  isinstance(action, CallUserAction):
+            raise Exception(f'Agent is stuck. Call user action executed. Here is the thought: {message.thought}')
+        
+        if isinstance(action, FinishedAction):
             return
+        
+        action.execute(self._agent_os)
```

Comment:
Beautiful refactoring :) 

Using pydantic for parsing + using the strategy pattern